src/helm.rs

```diff
@@ -347,16 +429,39 @@ pub fn add_repo(repo_name: &str, repo_url: &str) -> Result<(), HelmError> {
     Ok(())
 }
 
+/// Retrieves the Helm index file from the repository URL.
+#[instrument]
+pub async fn get_helm_index<T>(repo_url: T) -> Result<HelmRepo, HelmError>
+where
+    T: AsRef<str> + std::fmt::Debug,
+{
+    debug!("Get Helm repo index file");
+
+    let url = Url::parse(repo_url.as_ref()).context(UrlParseSnafu {})?;
+    let url = url.join(HELM_REPO_INDEX_FILE).context(UrlParseSnafu {})?;
+
+    debug!("Using {} to retrieve Helm index file", url);
+
+    let index_file_content = reqwest::get(url)
+        .await
+        .context(RequestSnafu {})?
+        .text()
+        .await
+        .context(RequestSnafu {})?;
+
+    serde_yaml::from_str(&index_file_content).context(YamlSnafu {})
+}
+
 /// Helper function to convert raw C string pointers to &str.
 fn ptr_to_str<'a>(ptr: *const i8) -> Result<&'a str, Utf8Error> {
```

Comment:
```suggestion
unsafe fn ptr_to_str<'a>(ptr: *const i8) -> Result<&'a str, Utf8Error> {
```

Reply:
It's unsafe to convert a raw pointer to a reference, since it's undefined behaviour unless you can guarantee that Rust's lifetime, null, and aliasing rules won't be violated for as long as the reference exists.

Reply:
Changed in df0076f.